gensim/test/test_fasttext.py

```diff
@@ -842,6 +842,176 @@ def test_sg_hs_against_wrapper(self):
         self.compare_with_wrapper(model_gensim, model_wrapper)
 
 
+with open(datapath('toy-data.txt')) as fin:
+    TOY_SENTENCES = [fin.read().strip().split(' ')]
+
+
+def train_gensim(min_count=5):
+    #
+    # Set parameters to match those in the load_native function
+    #
+    model = FT_gensim(bucket=100, size=5, alpha=0.05, workers=1, sample=0.0001, min_count=min_count)
+    model.build_vocab(TOY_SENTENCES)
+    model.train(TOY_SENTENCES, total_examples=len(TOY_SENTENCES), epochs=model.epochs)
+    return model
+
+
+def load_native():
+    #
+    # trained using:
+    #
+    # ./fasttext cbow -input toy-data.txt -output toy-model -bucket 100 -dim 5
+    #
+    path = datapath('toy-model.bin')
+    model = FT_gensim.load_fasttext_format(path)
+    return model
+
+
+class NativeTrainingContinuationTest(unittest.TestCase):
+    maxDiff = None
+
+    def test_in_vocab(self):
+        """Test for correct representation of in-vocab words."""
+        def yield_items(fin):
+            fin.readline()  # array shape
+            for line in fin:
+                columns = line.strip().split(' ')
+                word = columns.pop(0)
+                vector = [float(c) for c in columns]
+                yield word, np.array(vector, dtype=np.float32)
+
+        native = load_native()
+        with open(datapath('toy-model.vec')) as fin:
+            expected = dict(yield_items(fin))
+
+        for word, expected_vector in expected.items():
+            actual_vector = native.wv.word_vec(word)
+            self.assertTrue(np.allclose(expected_vector, actual_vector, atol=1e-4))
+
+    def test_out_of_vocab(self):
+        """Test for correct representation of out-of-vocab words."""
+        native = load_native()
+        #
+        # $ echo "quick brown fox jumps over lazy dog" | ./fasttext print-word-vectors gensim/test/test_data/toy-model.bin
+        #
+        expected = {
+            "quick": [0.023393, 0.11499, 0.11684, -0.13349, 0.022543],
+            "brown": [0.015288, 0.050404, -0.041395, -0.090371, 0.06441],
+            "fox": [0.061692, 0.082914, 0.020081, -0.039159, 0.03296],
+            "jumps": [0.070107, 0.081465, 0.051763, 0.012084, 0.0050402],
+            "over": [0.055023, 0.03465, 0.01648, -0.11129, 0.094555],
+            "lazy": [-0.022103, -0.020126, -0.033612, -0.049473, 0.0054174],
+            "dog": [0.084983, 0.09216, 0.020204, -0.13616, 0.01118],
+        }
+        expected = {word: np.array(arr, dtype=np.float32) for word, arr in expected.items()}
+
+        for word, expected_vector in expected.items():
+            actual_vector = native.wv.word_vec(word)
+            self.assertTrue(np.allclose(expected_vector, actual_vector, atol=1e-1))
```

Comment:
`atol=1e-1` not good, very suspicious, probably related with "hashing" issue (not sure)

Reply:
Opened https://github.com/RaRe-Technologies/gensim/issues/2315